diff --git a/portable-file-dialogs.h b/portable-file-dialogs.h
--- a/portable-file-dialogs.h
+++ b/portable-file-dialogs.h
@@ -1536,14 +1536,17 @@ inline notify::notify(std::string const &title,
         /* case icon::info: */ default: nid->dwInfoFlags = NIIF_INFO; break;
     }
 
-    ENUMRESNAMEPROC icon_enum_callback = [](HMODULE, LPCTSTR, LPTSTR lpName, LONG_PTR lParam) -> BOOL
+    struct icon_enum_callback
     {
-        ((NOTIFYICONDATAW *)lParam)->hIcon = ::LoadIcon(GetModuleHandle(nullptr), lpName);
-        return false;
+        static BOOL CALLBACK run(HMODULE, LPCTSTR, LPTSTR lpName, LONG_PTR lParam)
+        {
+            ((NOTIFYICONDATAW *)lParam)->hIcon = ::LoadIcon(GetModuleHandle(nullptr), lpName);
+            return FALSE;
+        }
     };
 
     nid->hIcon = ::LoadIcon(nullptr, IDI_APPLICATION);
-    ::EnumResourceNames(nullptr, RT_GROUP_ICON, icon_enum_callback, (LONG_PTR)nid.get());
+    ::EnumResourceNames(nullptr, RT_GROUP_ICON, &icon_enum_callback::run, (LONG_PTR)nid.get());
 
     nid->uTimeout = 5000;
 
